File: kinematic_robot/src/panda_joint_reader_node.cpp
//
// Created by rnm on 08.04.21.
//

#include <ros/ros.h>
#include "sensor_msgs/JointState.h"

void callback(const sensor_msgs::JointState& msg) {
    std::cout << "Joints: ";
    for (int i = 0; i < 7; ++i ) {
        std::cout << msg.position[i] << " ";
    }
    std::cout << "\n\n";
}

int main(int argc, char** argv)  {
    ros::init(argc, argv, "panda_joint_reader_node");

    // Get a node handle to the private parameters
    // e.g. /node_name/parameter_name can then be
    // read by nh.getParam("parameter_name", value);
    ros::NodeHandle  nh("~");

    // Parse parameters specified in the .launch file
    std::string topic_name;
    int queue_size;
    nh.getParam("topic_name", topic_name);
    nh.getParam("queue_size", queue_size);

    // Register a callback function (a function that is called every time a new message arrives)
    ros::Subscriber sub = nh.subscribe("/joint_states", 10, callback);

    // Prevent the program from terminating
    // Blocks but still allows messages to be received
    ros::spin();
}

File: kinematic_robot/scripts/Obsolete/panda_joint_reader_node.py
#!/usr/bin/python3

import rospy
import sys
from sensor_msgs.msg import JointState


def callback(msg):
    pos = msg.position
    print(pos)


def main(argv):
    # Initialize ROS connection
    # The script with the main function needs to be executable.
    # You can make the file executable by executing 'chmod +x /path/to/file.py'
    # in the terminal (without the "'")
    rospy.init_node("panda_joint_reader_node", argv)

    # Read parameters specified in the .launch file
    # The "~" in the beginning means that it will use parameters from the node
    # e.g. /node_name/parameter_name can then be
    # read by rospy.get_param("parameter_name", default)
    topic_name = rospy.get_param("~topic_name", "/default_topic_name")
    queue_size = rospy.get_param("~queue_size", 10)

    # Register a callback function (a function that is called every time a new message arrives)
    rospy.Subscriber(name=topic_name,
                     data_class=JointState,
                     callback=callback,
                     queue_size=queue_size)

    rospy.spin()


if __name__ == '__main__':
    main(sys.argv)

